snakemake_executor_plugin_googlebatch/executor.py

```diff
@@ -110,6 +110,19 @@ def validate(self, job):
                 f"Job {job} has container without image_family batch-cos*."
             )
 
+    def is_preemptible(self, job):
+        """
+        Determine if a job is preemptible.
+
+        The logic for determining if the set is valid should belong upstream.
+        """
+        is_p = self.workflow.remote_execution_settings.preemptible_rules.is_preemptible
+        if job.is_group():
+            preemptible = all(is_p(rule) for rule in job.rules)
```

Comment:
Not like this?
```suggestion
            preemptible = all(is_p(rule.name) for rule in job.rules)
```

Reply:
No, that's actually what I had first and it errored. The job.rules I think is a dict, so when we iterate like that we get the keys (the job names).

Reply:
It is an iterable of names.